docs/reference/ml/df-analytics/apis/put-inference.asciidoc

```diff
@@ -324,35 +260,160 @@ consequently, the probability of class 0 is `1 - P_1`. The class with the
+[%collapsible%open]
+========
+`weights`:::
 (Required, double) 
 The weights to multiply by the input values (the inference values of the trained 
 models).
+========
+//End weighted mode
 
 See <<ml-put-inference-aggregated-output-example>> for more details.
 
+=======
+//End aggregate output
+
+`classification_labels`::
+(Optional, string) 
+An array of classification labels.
+
+`feature_names`::
+(Optional, string)
+Features expected by the ensemble, in their expected order.
+
+`target_type`::
+(Required, string) 
+String indicating the model target type; `regression` or `classification.`
+
+`trained_models`::
+(Required, object)
+An array of `trained_model` objects. Supported trained models are `tree` and 
+`ensemble`.
+======
+//End ensemble
+
+See <<ml-put-inference-model-example>> for more details.
+
+=====
+//End trained model
+
+====
+//End definition
+
+`description`::
+(Optional, string) 
+A human-readable description of the {infer} trained model.
+
+//Begin inference_config
+`inference_config`::
+(Required, object)
+The default configuration for inference. This can be either a `regression`
+or `classification` configuration. It must match the underlying
+`definition.trained_model`'s `target_type`.
++
+.Properties of `inference_config`
+[%collapsible%open]
+====
+`regression`:::
+(Optional, object)
+Regression configuration for inference.
++
+.Properties of regression inference
+[%collapsible%open]
+=====
+`num_top_feature_importance_values`::::
+(Optional, integer)
+include::{docdir}/ml/ml-shared.asciidoc[tag=inference-config-regression-num-top-feature-importance-values]
+
+`results_field`::::
+(Optional, string)
+include::{docdir}/ml/ml-shared.asciidoc[tag=inference-config-results-field]
+=====
+
+`classification`:::
+(Optional, object)
+Classification configuration for inference.
++
+.Properties of classification inference
+[%collapsible%open]
+=====
+`num_top_classes`::::
+(Optional, integer)
+include::{docdir}/ml/ml-shared.asciidoc[tag=inference-config-classification-num-top-classes]
+
+`num_top_feature_importance_values`::::
+(Optional, integer)
+include::{docdir}/ml/ml-shared.asciidoc[tag=inference-config-classification-num-top-feature-importance-values]
+
+`results_field`::::
+(Optional, string)
+include::{docdir}/ml/ml-shared.asciidoc[tag=inference-config-results-field]
+
+`top_classes_results_field`::::
+(Optional, string)
+include::{docdir}/ml/ml-shared.asciidoc[tag=inference-config-classification-top-classes-results-field]
+=====
+====
+//End of inference_config
+
+//Begin input
+`input`::
+(Required, object) 
+The input field names for the model definition.
++
+.Properties of `input`
+[%collapsible%open]
+====
+`field_names`:::
+(Required, string) 
+An array of input field names for the model.
+====
+//End input
+
+`metadata`::
+(Optional, object) 
+An object map that contains metadata about the model.
+
+`tags`::
+(Optional, string) 
+An array of tags to organize the model.
+
 
 [[ml-put-inference-example]]
```

Comment:
IMO it's hard to figure out how all these partial examples fit together.  Could we provide one full example and then just link to https://github.com/elastic/ml-json-schemas/tree/master/examples for other combinations?